internal/api/grpc/idp/converter.go

```diff
@@ -659,13 +660,19 @@ func samlConfigToPb(providerConfig *idp_pb.ProviderConfig, template *query.SAMLI
 	if template.NameIDFormat.Valid {
 		nameIDFormat = nameIDToPb(template.NameIDFormat.V)
 	}
+	var metadataError *string
+	_, err := zsaml.ParseMetadata(template.Metadata)
```

Comment:
Why do we return the error on queries, when there shouldn't be an error anymore on command side? IMO I would remove this as well, as this could only lead to more confusion.